### PR TITLE
Arc - transactional observers should register standard synchronization instead of interposed

### PR DESCRIPTION
**Pull request by manovotn**
Arc-side fix for #18737

Cc @ochaloup

### CODE REVIEW

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java [hunk 1]
```diff
@@ -246,35 +247,45 @@ void notify(T event, ObserverExceptionHandler exceptionHandler, boolean async) {
...
+                                        || e.getCause() instanceof IllegalStateException
+                                        || e.getCause() instanceof SystemException) {
+                                    // registration failed, AFTER_SUCCESS OMs are accordingly to CDI spec left out
+                                    predicate = predicate.and(this::isNotAfterSuccess);
+                                }
                             }
                         }
+                    } catch (SystemException e) {
+                        // In theory, this can be thrown by TransactionManager#getStatus() at which point we cannot even
+                        // determine if we should register some synchronization, therefore, no action is taken
```
Comment (manovotn) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java:288]: No entirely sure about this, but this seemed like the most reasonable approach to me. Comments are welcome :)

Comment (ochaloup) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java:288]: Could not be added some debug logging that would print the exception with stacktrace, to be able to find the root cause on investigation?

**Review by ochaloup — APPROVED** [review]
To me all is fine. Just added a note under the existing comment.

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java (hunk 1 above)
Comment (manovotn) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java:288]: Good point, can add that tomorrow.

**Comment by mkouba** [comment]
Some CI failures are probably related..

**Comment by manovotn** [comment]
Yeah, I know. Apparently, JTA already has a CDI bean for `TransactionManager`. Fixing that now.

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java (hunk 1 above)
Comment (manovotn) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java]: Fixed this, thanks for pointing it out.

**Comment by ochaloup** [comment]
@manovotn oh, sorry I haven't realized that we have provided the CDI bean for TM in JTA already...

**Comment by manovotn** [comment]
No worries, at least next time I'll know :-)

**Comment by manovotn** [comment]
These failures no longer look relevant, they're coming from Hibernate multitenancy tests.

@mkouba please review this PR when you have a moment to spare

**Review by mkouba — APPROVED** [review]
I can't speak about the JTA part but the rest of the PR looks good! ;-)

**Comment by Sanne** [comment]
hey, the Hibernate ORM / multitenancy integration tests started failing today. I suspect it's related?

```
CDI: programmatic lookup problem detected
-----------------------------------------
At least one bean matched the required type and qualifiers but was marked as unused and removed during build
Removed beans:
	- PRODUCER_METHOD bean io.quarkus.narayana.jta.runtime.NarayanaJtaProducers#transactionSynchronizationRegistry() [types=[interface javax.transaction.TransactionSynchronizationRegistry], qualifiers=null]
Required type: interface javax.transaction.TransactionSynchronizationRegistry
Required qualifiers: [@javax.enterprise.inject.Default()]
```

Please see https://github.com/quarkusio/quarkus/pull/19928/checks?check_run_id=3524105980

**Pull request merged**